### ISSUE TITLE
REPL tab completion exception

### ISSUE THREAD

**Issue by bkamins** [JuliaLang/julia#19310]
Under Windows opening a new terminal and writing `"C` and then pressing *tab* key raises an exception:

```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.5.0 (2016-09-19 18:14 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-w64-mingw32

julia> "CWARNING: Caught an exception in the keymap:
ERROR: stat: operation not permitted (EPERM)
ERROR: syntax: incomplete: invalid string syntax

julia> pwd()
"D:\\"

julia>
```

This happens only if current directory is drive root, e.g. `C:\\`, `D:\\`. It might be related to #8693.

**Comment by dhoegh** [comment]
This is fixed in #18567. Unfortunately the actions to get the PR merged are unclear.

**Comment by musm** [comment]
This is now fixed (probably by https://github.com/JuliaLang/julia/pull/22531)

**Comment by tkelman** [comment]
22531 was not merged. is this tested?

**Comment by StefanKarpinski** [comment]
There's a test in #22531, does that not suffice?

**Comment by tkelman** [comment]
There's no ampersand here, so #22531 seems to be fixing a different issue? It also doesn't have the part of the test at https://github.com/JuliaLang/julia/pull/18567/files#diff-51b80b16c8d771c4166824cd1c15a66cR679 which looks closer to this issue.

**Comment by dhoegh** [comment]
This issue is caused by `isfile` can throw an error. In #18567 I wrapped all `isfile` calls in `try`, but @keno pointed out it was not the correct solution in comment https://github.com/JuliaLang/julia/pull/18567#issuecomment-253626970. 
If all agree that `isfile` should be non throwing then this issue is resolved be fixing `isfile`. #22531 do not include the fix for this issue, but only for #18479.
